### ISSUE TITLE
No Error when two defaultprops are used for the same component

### ISSUE THREAD

**Issue by alexd99** [yannickcr/eslint-plugin-react#2480]
I am on a team have that is using this plugin, and we recently ran into an issue where someone added two `defaultprops` to the same component, and eslint did not throw an error. I am wondering if this could be added to the recommended settings? 

**eslint-plugin-react verision**
 7.16.0

**.eslintrc.json**
```text
{
  "env": {
    "node": true,
    "browser": true,
    "jest": true,
    "es6": true
  },
  "parserOptions": {
    "ecmaVersion": 2018
  },
  "parser": "babel-eslint",
  "plugins": ["import", "react", "jsx-a11y", "react-hooks"],
  "extends": [
    "eslint:recommended",
    "plugin:import/errors",
    "plugin:import/warnings",
    "plugin:react/recommended",
    "plugin:jsx-a11y/recommended"
  ],
  "rules": {
    "react-hooks/rules-of-hooks": "error",
    "react-hooks/exhaustive-deps": "error",
    "no-unused-vars": [
      "error",
      {
        "ignoreRestSiblings": true
      }
    ],
    "import/namespace": [
      "error",
      {
        "allowComputed": true
      }
    ],
    "react/no-find-dom-node": "off"
  }
}
```

**Comment by ljharb** [comment]
It would be an error if it were static class fields; there’s generally not any linter rule that governs duplicate imperative mutation of the same property, and I’m not convinced it’s a good rule to have.

Since you’re using babel-eslint, why are you not using static class fields for defaultProps?

**Comment by alexd99** [comment]
The component is a functional component that is wrapped in a `React.forwardRef` as shown below.
```jsx
const Icon = React.forwardRef(buttonSwitcher(Link, Button));

Icon.defaultProps = { type: "button" };

Icon.defaultProps = { size: "normal" };
```
Since it is not a class and we can't destructure props the only option we have available is to use Icon.defaultProps`.

**Comment by trevordmiller** [comment]
I've seen this problem as well. Some tools like `react-docgen` need to use `SomeFuctionalComponent.someField` to be able to parse properly.

**Comment by ljharb** [comment]
defaultProps on functional components are being deprecated - i wasn't aware forwardRef even supported defaultProps. 

Regardless, I'd expect the thing *being wrapped* should have defaultProps if it needs them, not the wrapper.
